[Fuzzer Binaries] Add wgslfuzz to WebGPU build

JonWBedard opened the pull request:
#### 6745d4faafa9b91af689db886e9089d09bab1266
<pre>
[Fuzzer Binaries] Add wgslfuzz to WebGPU build
<a href="https://bugs.webkit.org/show_bug.cgi?id=282258">https://bugs.webkit.org/show_bug.cgi?id=282258</a>
<a href="https://rdar.apple.com/138854751">rdar://138854751</a>

Reviewed by Elliott Williams and Mike Wyrzykowski.

* Configurations/WebKitProjectPaths.xcconfig: Add WEBKIT_FUZZERS_DIR.
* Source/WebGPU/Configurations/wgslfuzz.xcconfig: Install wgslfuzz when ENABLE_LIBFUZZER is true.
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj: Build wgslfuzz with WebGPU.

Canonical link: <a href="https://commits.webkit.org/285909@main">https://commits.webkit.org/285909@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d3239f9553e2b5ed02cce07bd74afda2fa183642

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/74136 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/53565 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/26947 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/78509 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/25374 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/62698 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/1350 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/58289 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/16631 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/77203 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/48444 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/63773 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/38700 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/45352 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/23707 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/66825 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/21615 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/80028 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/1453 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/807 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/66591 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/1597 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/63790 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/65864 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16335 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/9793 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/7953 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/1417 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/1446 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/1434 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/1453 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/WebGPU/Configurations/wgslfuzz.xcconfig:
```diff
@@ -25,4 +25,8 @@ PRIVATE_HEADERS_FOLDER_PATH = $(WGSL_INSTALL_PATH_PREFIX)$(WK_LIBRARY_HEADERS_FO
 SYSTEM_HEADER_SEARCH_PATHS = $(SDK_DIR)$(WK_ALTERNATE_WEBKIT_SDK_PATH)$(WK_LIBRARY_HEADERS_FOLDER_PATH) $(inherited);
+
+SKIP_INSTALL = $(WK_NOT_$(ENABLE_LIBFUZZER));
+INSTALL_PATH = $(WEBKIT_FUZZERS_DIR)
```
nit: project style is to use a trailing semicolon

emw-apple commented on Source/WebGPU/Configurations/wgslfuzz.xcconfig:
```diff
@@ -25,4 +25,8 @@ PRIVATE_HEADERS_FOLDER_PATH = $(WGSL_INSTALL_PATH_PREFIX)$(WK_LIBRARY_HEADERS_FO
 SYSTEM_HEADER_SEARCH_PATHS = $(SDK_DIR)$(WK_ALTERNATE_WEBKIT_SDK_PATH)$(WK_LIBRARY_HEADERS_FOLDER_PATH) $(inherited);
+
+SKIP_INSTALL = $(WK_NOT_$(ENABLE_LIBFUZZER));
+INSTALL_PATH = $(WEBKIT_FUZZERS_DIR)
```
I might prefer that we install this to the WebGPU.framework, if the location is flexible. No need to start installing to a  brand new directory. Here's what we do in WebKitTestRunner: https://github.com/WebKit/WebKit/blob/fe663b06e8da7388f053104ca13de9ca48730212/Tools/WebKitTestRunner/Configurations/BaseTarget.xcconfig#L24-L27

ddkilzer replied:
I understand you want to put the binary with the associated framework as a resource, but doing this makes it difficult to see a list of all the WebKit-project-related fuzzers in one place.

If we did this consistently, we'd put all of the WebCore fuzzers in `WebCore.framework/Resources`, but then where would the `libANGLE-dynamic` and `libwebrtc` fuzzers go since they are dylibs?

I guess I'd rather see them all go to one place first.

> No need to start installing to a brand new directory.

One of the main points of this PR is to install the fuzzers to a specific, well-defined location.

JonWBedard committed:
[Fuzzer Binaries] Add wgslfuzz to WebGPU build
https://bugs.webkit.org/show_bug.cgi?id=282258
rdar://138854751

Reviewed by Elliott Williams and Mike Wyrzykowski.

* Configurations/WebKitProjectPaths.xcconfig: Add WEBKIT_FUZZERS_DIR.
* Source/WebGPU/Configurations/wgslfuzz.xcconfig: Install wgslfuzz when ENABLE_LIBFUZZER is true.
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj: Build wgslfuzz with WebGPU.

Canonical link: https://commits.webkit.org/285909@main
